src/dex/synthetix/synthetix.ts

```diff
@@ -0,0 +1,334 @@
+import _ from 'lodash';
+import { ethers } from 'ethers';
+import {
+  Token,
+  Address,
+  ExchangePrices,
+  AdapterExchangeParam,
+  SimpleExchangeParam,
+  PoolLiquidity,
+  Logger,
+} from '../../types';
+import { SwapSide, Network, NULL_ADDRESS } from '../../constants';
+import { getBigIntPow, getDexKeysWithNetwork, _require } from '../../utils';
+import { IDex } from '../../dex/idex';
+import { IDexHelper } from '../../dex-helper/idex-helper';
+import { DexParams, SynthetixData } from './types';
+import { SimpleExchange } from '../simple-exchange';
+import { SynthetixConfig, Adapters } from './config';
+import { Interface } from '@ethersproject/abi';
+import { MultiWrapper } from '../../lib/multi-wrapper';
+import {
+  OPTIMISM_STATE_TTL_IN_S,
+  SYNTHETIX_GAS_COST_WITHOUT_SUSD,
+  SYNTHETIX_GAS_COST_WITH_SUSD,
+} from './constants';
+import { synthetixMath } from './contract-math/synthetix-math';
+import { SynthetixState } from './synthetix-state';
+// There are so many ABIs, where I need only one or two functions
+// So, I decided to unite them into one combined interface
+import CombinedSynthetixABI from '../../abi/synthetix/CombinedSynthetix.abi.json';
+
+export class Synthetix extends SimpleExchange implements IDex<SynthetixData> {
+  readonly hasConstantPriceLargeAmounts = false;
+  readonly needWrapNative = true;
+
+  readonly combinedIface: Interface;
+
+  readonly multiWrapper: MultiWrapper;
+
+  logger: Logger;
+
+  synthetixState: SynthetixState;
+
+  public static dexKeysWithNetwork: { key: string; networks: Network[] }[] =
+    getDexKeysWithNetwork(SynthetixConfig);
+
+  constructor(
+    readonly network: Network,
+    protected dexKey: string,
+    readonly dexHelper: IDexHelper,
+    protected adapters = Adapters[network] || {},
+    readonly config = SynthetixConfig[dexKey][network],
+  ) {
+    super(dexHelper.config.data.augustusAddress, dexHelper.web3Provider);
+    this.config = this.normalizeConfig(this.config);
+    this.logger = dexHelper.getLogger(dexKey);
+    this.combinedIface = new Interface(CombinedSynthetixABI);
+    this.multiWrapper = new MultiWrapper(
```

Comment:
We should probably instantiate one by chain

Reply:
Yeah, you are right. Fixed